packages/driver/src/multi-domain/cypress.ts

```diff
@@ -30,6 +32,19 @@ const createCypress = () => {
     setup(config, env)
   })
 
+  Cypress.specBridgeCommunicator.on('generate:final:snapshot', (snapshotUrl: string) => {
+    const specBridgeLocation = $Location.create(window.location.href)
+    const requestedSnapshotUrlLocation = $Location.create(snapshotUrl)
+
+    if (requestedSnapshotUrlLocation.originPolicy === specBridgeLocation.originPolicy) {
+      // if true, this is the correct specbridge to take the snapshot and send it back
+      // @ts-ignore
+      const finalSnapshot = cy.createSnapshot(FINAL_SNAPSHOT_NAME)
```

Comment:
Maybe add this to the internal types?

Reply:
added in [9ef3f56](https://github.com/cypress-io/cypress/pull/20949/commits/9ef3f56fd6bce69a56d98ae42e981b352ff52a3e). The type itself isn't incredibly useful IMO but that seems to be the current convention in the Cy internal types.